bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/link/LinkImpl.java

```diff
@@ -90,10 +78,30 @@ public boolean isValid() {
      * @return Link URL, can be {@code null} if link is not valid
      */
     @Override
+    @JsonIgnore
     public @Nullable String getURL() {
         return url;
     }
 
+    /**
+     * Getter for the processed URL.
+     *
+     * @return Processed link URL, can be {@code null} if link is not valid or no processors are defined
+     */
+    @Override
+    @JsonProperty("url")
+    public @Nullable String getProcessedURL() {
+        if (!processed) {
+            if (linkProcessor != null) {
+                processedUrl = linkProcessor.process(url);
+            } else {
+                processedUrl = null;
```

Comment:
in case there is no processor, shouldn't it return the plain URL (the output of `getURL()`)?

Reply:
Yes, good catch! Fixed in aae5395